src/atomic.cr

```diff
@@ -283,6 +291,8 @@ struct Atomic(T)
     {% elsif T.union_types.all? { |t| t == Nil || t < Reference } && T != Nil %}
       address = atomicrmw(:xchg, pointerof(@value).as(LibC::SizeT*), LibC::SizeT.new(value.as(Void*).address), ordering)
       Pointer(T).new(address).as(T)
+    {% elsif T == Bool %}
+      cast_from atomicrmw(:xchg, as_pointer, cast_to(value), ordering)
```

Comment:
**thought:** I'm wondering if we could expand the refactor with `cast_from` and `cast_to` to the other branches here as well. Looks like the resulting code would be identical in all, simplifying this def.

Reply:
Other branches don't need the casts, but the `else` branch could, and then we wouldn't need a special case for Bool :+1: